core/templates/dev/head/pages/exploration_player/PlayerTranscriptService.js

```diff
@@ -136,13 +118,13 @@ oppia.factory('PlayerTranscriptService', [
         }
         return transcript[index];
       },
-      getLastAnswerOnActiveCard: function(index) {
+      getLastAnswerOnActiveCard: function(activeCardIndex) {
         if (
-          this.isLastCard(index) ||
-          transcript[index].getStateName() === null) {
+          this.isLastCard(activeCardIndex) ||
+          transcript[activeCardIndex].getStateName() === null) {
           return null;
         } else {
-          return transcript[index].
+          return transcript[activeCardIndex].
             getInputResponsePairs().slice(-1)[0].learnerInput;
```

Comment:
Btw, sometimes we get the production error "Cannot find property learnerInput of undefined" originating here. Could we guard against this? Not sure when it happens, perhaps in EndExploration or Continue or something (but it has occurred).

Could we perhaps return null here when there are no inputResponsePairs()? (And add a test to check this too.)

Thanks!

Reply:
Done, added tests also.